common/src/main/java/com/wynntils/services/iterator/IteratorService.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.iterator;
+
+import com.wynntils.core.components.Service;
+import com.wynntils.mc.event.TickEvent;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class IteratorService extends Service {
```

Comment:
It's kind of weird to have a complete service for this... otoh, I have no better suggestion at the moment. We'll have to be on the lookout for some suitable functionality to group it with.

Reply:
Just like the function this name is too related to programming.

Reply:
All this code in this service.. I feel like this is a ticking bomb. If the only use case of an iterator function is coloring, can we just simplify this class?

Reply:
If it were not for the onTick, I would have suggested putting all of this in the function itself. It is a very special thing that is only needed for showing rainbow colors in info boxes. I fear that Donkey might have made this too generic which only confuses things.

Reply:
Actually, the function can probably use the TickManager instead. A bit more overhead, but hey, we can live with that.

Reply:
I actually wanted it to be generic. Since you can pass in strings and concat them with `[` `]`, you can get one area in your screen to change the actual text it's displaying. I don't have an exact idea or use case for this in mind, but generally the community is pretty creative and might find it useful. After all I did not expect them to use soul point timers or mempct to change colors of text.